modules/xml/filterx-func-format-windows-eventlog-xml.c

```diff
@@ -54,7 +54,7 @@ _append_inner_data_dict_element(FilterXObject *key, FilterXObject *value, gpoint
     }
 
   if (value_str_len)
-    g_string_append_printf(buffer, "<Data Name='%s'>%s</Data>", key_str, value_str);
+    g_string_append_printf(buffer, "<Data Name='%s'>%s</Data>", key_str, g_markup_escape_text(value_str, value_str_len));
```

Comment:
All `g_markup_escape_text()` invocations leak here, we need to pass ownership where possible, and deallocate the memory otherwise.